File: sdks/python/examples/model_management.py
#!/usr/bin/env python3
"""
Model management example using MLXR Python SDK.

Demonstrates listing, pulling, and managing models.
"""

from mlxrunner import MLXR

def main():
    client = MLXR()

    print("Model Management Example")
    print("=" * 50)

    # List models (OpenAI API)
    print("\n1. List models (OpenAI API):")
    models = client.models.list()
    for model in models.data:
        print(f"  - {model.id}")

    # List models (Ollama API)
    print("\n2. List models (Ollama API):")
    ollama_models = client.ollama.list()
    for model in ollama_models.models:
        print(f"  - {model.name} ({model.size} bytes)")

    # List running models
    print("\n3. List running models:")
    running = client.ollama.ps()
    if running.models:
        for model in running.models:
            print(f"  - {model.name} (expires: {model.expires_at})")
    else:
        print("  No models currently running")

    # Pull a model (streaming)
    print("\n4. Pull a model (example - commented out):")
    print("  # Uncomment to actually pull:")
    print("  # for chunk in client.ollama.pull('llama2:7b', stream=True):")
    print("  #     print(f'  Status: {chunk.status}')")

    # Show model info
    print("\n5. Show model info (example - commented out):")
    print("  # Uncomment if you have a model:")
    print("  # info = client.ollama.show('TinyLlama-1.1B')")
    print("  # print(f'  Family: {info.details.family}')")
    print("  # print(f'  Parameters: {info.details.parameter_size}')")

    client.close()


if __name__ == "__main__":
    main()

File: sdks/typescript/examples/model-management.ts
/**
 * Example: Model management with Ollama API
 */

import { MLXRClient } from '../src';

async function main() {
  const client = new MLXRClient();

  console.log('MLXR TypeScript SDK - Model Management Example\n');

  // List all models
  console.log('1. List all models:');
  try {
    const tags = await client.ollama.tags();
    console.log('Available models:', tags.models.length);
    tags.models.forEach((model) => {
      console.log(`  - ${model.name}`);
      console.log(`    Size: ${(model.size / 1024 / 1024).toFixed(2)} MB`);
      console.log(`    Modified: ${model.modified_at}`);
    });
  } catch (error) {
    console.error('Error:', error);
  }

  // Show model details
  console.log('\n2. Show model details:');
  try {
    const details = await client.ollama.show({ name: 'tinyllama' });
    console.log('Model details:');
    console.log('  Format:', details.details.format);
    console.log('  Family:', details.details.family);
    console.log('  Parameters:', details.details.parameter_size);
    console.log('  Quantization:', details.details.quantization_level);
  } catch (error) {
    console.error('Error:', error);
  }

  // List running models
  console.log('\n3. List running models:');
  try {
    const running = await client.ollama.ps();
    console.log('Running models:', running.models.length);
    running.models.forEach((model) => {
      console.log(`  - ${model.name}`);
      console.log(`    VRAM: ${((model.size_vram || 0) / 1024 / 1024).toFixed(2)} MB`);
    });
  } catch (error) {
    console.error('Error:', error);
  }

  // Pull a model (with streaming progress)
  console.log('\n4. Pull a model:');
  try {
    console.log('Pulling tinyllama (this may take a while)...');
    const stream = client.ollama.streamPull({
      name: 'tinyllama',
      stream: true,
    });

    for await (const progress of stream) {
      if (progress.total && progress.completed) {
        const percent = ((progress.completed / progress.total) * 100).toFixed(1);
        process.stdout.write(`\r  Progress: ${percent}%`);
      } else {
        console.log(`  Status: ${progress.status}`);
      }
    }
    console.log('\n  Model pulled successfully!');
  } catch (error) {
    console.error('Error:', error);
  }
}

main().catch(console.error);
